task/aws/task.go

```diff
@@ -221,7 +221,9 @@ func (t *Task) Delete(ctx context.Context) error {
 			}
 		}
 		logrus.Info("[2/8] Emptying Bucket...")
-		if err := machine.Delete(ctx, (*t.DataSources.Credentials.Resource)["RCLONE_REMOTE"]); err != nil && err != common.NotFoundError {
+
+		// TODO: t.DataSources.Credentials.Resource may be nil, check first.
```

Comment:
Probably a dumb question, but in which case[s] may `t.DataSources.Credentials.Resource` be `<nil>`? :think:

Reply:
I bulk added that comment when before I refactored `Credentials.Resource` to not be a pointer to a map, forgot to remove it.

In this case `Resource` can be nil if the data source's `Read` returns an error or is never called. If those were write attempts I'd add `nil` checks here anyway, since intialization of the map is significantly distant from this code for it to be considered external. But go's maps are read-safe even when unallocated, so we're fine.